clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java

```diff
@@ -162,8 +167,11 @@ public void start(final ServiceProvider<Service> serviceProvider) {
     } catch (RuntimeException e) {
       entityFactory = null;
       try {
-        clusterConnection.close();
+        Connection conn = clusterConnection;
```

Comment:
Yes I can extract then. The idea is to nullify as soon as possible the connection so that isConnected() returns false, because we can have some scheduled threads that will call `isConnected()`. So if the `close()` takes some time for any reason, this is to avoid a scheduled thread to still see that we are connected and thus try to send some messages.